hazelcast/src/main/java/com/hazelcast/internal/cluster/fd/FailureDetector.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright (c) 2008-2017, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.internal.cluster.fd;
+
+/**
+ * Failure detector tracks heartbeats of a member and decides liveness/availability of the member.
+ */
+public interface FailureDetector {
```

Comment:
`FailureDetector` tracks only a single member but `ClusterFailureDetector` tracks multiple members (cluster). So a `ClusterFailureDetector` can be formed by multiple `FailureDetector`s.